### PR TITLE
Analytics - integration of Amplitude 

### PR DESCRIPTION
**Pull request by timhaines**

### CODE REVIEW

**Comment by timhaines** [comment]
This is still WIP.  Not yet ready for review.

**Comment by timhaines** [comment]
Ready for review.  🍔

File: app/components/stripe-checkout.js [hunk 1]
```diff
@@ -62,6 +62,13 @@ export default Ember.Component.extend({
       var chosenPlan = this.get('plan');
       var planName = this.get('planName');
 
+      let organizationId = this.get('organization.id')
```
Comment (fotinakis) [inline app/components/stripe-checkout.js]: Need semi-colon here.

File: app/routes/organization/project/builds/build.js [hunk 2]
```diff
@@ -22,6 +22,16 @@ export default Ember.Route.extend(AuthenticatedRouteMixin, {
     }
   },
   actions: {
+    didTransition() {
+      let organizationId = this.modelFor('organization').get('id');
+      let build = this.modelFor(this.routeName);
+      let eventProperties = {
+        project_id: build.get('project.id'),
```
Comment (fotinakis) [inline app/routes/organization/project/builds/build.js]: Also include `project_slug: build.get('project.slug')` for consistency with backend build events.

File: app/routes/organizations/organization/billing.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import Ember from 'ember';
+import AuthenticatedRouteMixin from 'ember-simple-auth/mixins/authenticated-route-mixin';
+
+export default Ember.Route.extend(AuthenticatedRouteMixin, {
+  actions: {
+    didTransition() {
+      let organization = this.modelFor(this.routeName);
+      let eventProperties = {};
```
Comment (fotinakis) [inline app/routes/organizations/organization/billing.js]: Drop this line, make `track` default to {} when no properties are given.

Also per discussion, flip arg ordering to be `event`, `organization` (not orgId), `eventProperties`.

File: app/routes/pricing.js [hunk 4]
```diff
@@ -2,4 +2,54 @@ import Ember from 'ember';
 import ResetScrollMixin from '../mixins/reset-scroll';
 
 export default Ember.Route.extend(ResetScrollMixin, {
+  actions: {
+    didTransition() {
+      // Track Pricing Viewed without the organization for now.
+      this.analytics.track(null, 'Pricing Viewed', {});
+
+      // TODO: Track with the last organization
```
Comment (fotinakis) [inline app/routes/pricing.js]: Let's drop all of this for now out of this PR, for cleanliness. Maybe push this branch to another branch to keep this around for reference, but this seems like code we'd either never update / will drop entirely / or will re-write entirely so we should drop this now.

File: config/environment.js [hunk 5]
```diff
@@ -44,6 +44,12 @@ module.exports = function(environment) {
   ENV.APP.INTERCOM_APP_ID = 'itdwmqct';
   ENV.APP.HEAP_APP_ID = '3430193657';
 
+  ENV.APP.AMPLITUDE_USERS_INSTANCE_NAME = 'Users';
+  ENV.APP.AMPLITUDE_USERS_PROJECT_ID = '5a8c0499760103fcd2754fe7d5756214';
+  ENV.APP.AMPLITUDE_ORGANIZATIONS_INSTANCE_NAME = 'Organizations';
+  ENV.APP.AMPLITUDE_ORGANIZATIONS_PROJECT_ID = '89f8fae9aab3fccc0740237f17e43745';
+
+
```
Comment (fotinakis) [inline config/environment.js]: Add the prod env vars in the `environment == 'production'` block.

File: app/services/analytics.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    if(organizationId) {
+      userEventProperties = Object.assign(
+        {'organization_id': organizationId},
+        eventProperties
+      );
+    }
+
+    this.userInstance.logEvent(eventName, userEventProperties);
+
+    if(organizationId) {
```
Comment (fotinakis) [inline app/services/analytics.js]: Space between `if` and the conditional, here and elsewhere. I'll update the eslint.

File: app/services/analytics.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+      this.organizationInstance.setUserId(organizationId);
```
Comment (fotinakis) [inline app/services/analytics.js]: I also think we should call `setGroup('organization_id', organizationId)` to mirror backend behavior and at least store the data for potential future use.

File: app/components/stripe-checkout.js [hunk 8]
```diff
@@ -62,6 +62,13 @@ export default Ember.Component.extend({
...
+      let eventProperties = {
+        plan_id: chosenPlan,
+        current_plan_id: this.get('organization.subscription.plan'),
+      };
+      this.analytics.track(organizationId, 'Billing Plan Selected', eventProperties);
```
Comment (fotinakis) [inline app/components/stripe-checkout.js]: Move this into `else` below.

Comment (timhaines) [inline app/components/stripe-checkout.js]: We decided to wrap this in if (!this.get('updateCard')) instead.

**Comment by timhaines** [comment]
@fotinakis Have addressed all of your comments.  Ready for re-review.

File: app/routes/organization/project/index.js [hunk 9]
```diff
@@ -3,5 +3,14 @@ import AuthenticatedRouteMixin from 'ember-simple-auth/mixins/authenticated-rout
 import ResetScrollMixin from 'percy-web/mixins/reset-scroll';
 
 export default Ember.Route.extend(AuthenticatedRouteMixin, ResetScrollMixin, {
+  actions: {
+    didTransition() {
+      let project = this.modelFor(this.routeName);
+      let organization = project.get('organization');
+      let eventProperties = {
+        project_id: project.get('id')
```
Comment (fotinakis) [inline app/routes/organization/project/index.js]: Add `project_slug`

File: app/routes/organization/index.js [hunk 10]
```diff
@@ -8,5 +8,10 @@ export default Ember.Route.extend(AuthenticatedRouteMixin, {
   afterModel(model) {
     model.get('projects').reload();
   },
+  actions: {
+    didTransition() {
+      let organization = this.modelFor(this.routeName);
+      this.analytics.track('Dashboard Viewed', organization, {});
```
Comment (fotinakis) [inline app/routes/organization/index.js]: Drop `{}`

File: app/routes/organization/project/builds/build.js [hunk 11]
```diff
@@ -22,6 +22,17 @@ export default Ember.Route.extend(AuthenticatedRouteMixin, {
     }
   },
   actions: {
+    didTransition() {
+      let organization = this.modelFor('organization');
```
Comment (fotinakis) [inline app/routes/organization/project/builds/build.js]: Per discussion, drop this, use `build.get('project.organization')`

**Comment by fotinakis** [comment]
LGTM 🚀 🎉  👍 👍 👍  Thanks for this!

**Pull request merged**